tests/PHPStan/Analyser/data/bug-12949.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Bug12949;
+
+function doFoo():void {
+	$b = '0';
+	${$b} = 1;
+
+	echo "";
+}
+
+function doBar(object $o):void {
+	$b = '0';
+	$o->{$b}();
+	$o::{$b}();
+	echo $o::{$b};
```

Comment:
I don't think these cases are realistic, but I fixed them to keep the implementation of the 4 rules in sync